osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -243,5 +235,32 @@ public DaySeparator(DateTimeOffset time)
                 };
             }
         }
+
+        /// <summary>
+        /// An <see cref="OsuScrollContainer"/> with functionality to automatically scrolls whenever the maximum scrollable distance increases.
+        /// </summary>
+        private class ChannelScrollContainer : OsuScrollContainer
+        {
+            private const float auto_scroll_leniency = 10f;
+
+            private float? lastExtent;
+
+            /// <summary>
+            /// Whether this should automatically scroll to end on the next call to <see cref="UpdateAfterChildren"/>.
```

Comment:
```suggestion
            /// Whether this container should automatically scroll to end on the next call to <see cref="UpdateAfterChildren"/>.
```

but why is this public? feels quite convoluted structure wise...

Reply:
it's unfortunately required to be public because of the scrolling adjustment from old messages removal logic, as explained in the usage:
https://github.com/ppy/osu/blob/e806e5bcd1580f1c85026e53d289d7cf933e4767/osu.Game/Overlays/Chat/DrawableChannel.cs#L127-L146
https://github.com/ppy/osu/blob/e806e5bcd1580f1c85026e53d289d7cf933e4767/osu.Game/Overlays/Chat/DrawableChannel.cs#L148-L151